<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>628</width>
    <height>614</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>pksvm_gui</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>2</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Training</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QGridLayout" name="gridLayout">
          <item row="2" column="0">
           <widget class="QLabel" name="label_2">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Attribute name of the class label in the training vector file (default is label)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>attribute name for class label</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QToolButton" name="toolButton_training">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_4">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;OGR supported vector file. Features must contain labels and band information. You can use pkextract to prepare training vector files&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Training vectorfile</string>
            </property>
           </widget>
          </item>
          <item row="3" column="2">
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="training"/>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="nclass"/>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_20">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Provide number of classes in training file here to prepare class name-value pairs (push button prepare table). This is mandatory if labels in training vector are provided as text (instead of Integer values). The table also provide priors and balance classes in training sample.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>number of classes to prepare table</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="label"/>
          </item>
          <item row="2" column="2">
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_21">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Name of the training layer (optional). If not provided, all layers will be read per default.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Training layername (optional)</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="tln"/>
          </item>
          <item row="1" column="2">
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QCommandLinkButton" name="commandLinkButtonPrepareTable">
          <property name="text">
           <string>prepare table</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLabel" name="label_17">
            <property name="text">
             <string>class name-value pairs</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTableView" name="tableView_labels"/>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Input/Output</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="2" column="5">
           <widget class="QLineEdit" name="nodata"/>
          </item>
          <item row="1" column="4">
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>msknodata</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QToolButton" name="toolButton_input">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="2" column="4">
           <widget class="QLabel" name="label_16">
            <property name="text">
             <string>nodata</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_13">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Dimensions of mask image must correspond to input image. Pixels in mask with values=masknodata are not classified and get nodata value.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Mask image</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QToolButton" name="toolButton_mask">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_15">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Output image (classified map). Can be OGR vector or GDAL raster, depending on input data.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Output data</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="mask"/>
          </item>
          <item row="1" column="5">
           <widget class="QLineEdit" name="msknodata"/>
          </item>
          <item row="1" column="3">
           <spacer name="horizontalSpacer_15">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="output"/>
          </item>
          <item row="2" column="2">
           <widget class="QToolButton" name="toolButton_output">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="input"/>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_3">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Input image. Both OGR vectors and GDAL rasters are supported. Number of bands must match those in training sample.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Input data</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_22">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Color table must be provided as ASCII text file with five (space delimited) columns:&lt;/p&gt;&lt;p&gt;value red green blue alpha&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>color table</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="ct"/>
          </item>
          <item row="3" column="2">
           <widget class="QToolButton" name="toolButton_ct">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Classifier</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="3" column="1">
           <widget class="QLineEdit" name="cv"/>
          </item>
          <item row="2" column="4">
           <widget class="QLineEdit" name="ccost"/>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_5">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Leave empty to skip cross validation. Enter 2 for a two-fold cross validation&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>n-fold cross validation</string>
            </property>
           </widget>
          </item>
          <item row="2" column="3">
           <widget class="QLabel" name="label_10">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;the parameter C of C_SVC, epsilon_SVR, and nu_SVR (default is 1)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>CCost</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QLabel" name="label_11">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Coef0 in the kernel function (default is 0)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Coef 0</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_6">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Type of the support vector machine&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>SVM type</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="svmtype"/>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="gamma"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_7">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;type of kernel function (default is radial)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Kernel type</string>
            </property>
           </widget>
          </item>
          <item row="0" column="4">
           <widget class="QLineEdit" name="coef0"/>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_9">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;gamma in kernel function (default is 0)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Gamma</string>
            </property>
           </widget>
          </item>
          <item row="0" column="7">
           <widget class="QLineEdit" name="nu"/>
          </item>
          <item row="1" column="4">
           <widget class="QLineEdit" name="kd"/>
          </item>
          <item row="0" column="6">
           <widget class="QLabel" name="label_12">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;the parameter nu of nu_SVC, one_class SVM, and nu_SVR (default is 0.5)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>nu</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <spacer name="verticalSpacer_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="kerneltype"/>
          </item>
          <item row="1" column="3">
           <widget class="QLabel" name="label_8">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;degree in kernel function (default is 3)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Kernel degree</string>
            </property>
           </widget>
          </item>
          <item row="0" column="5">
           <spacer name="horizontalSpacer_5">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>Console</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="QLabel" name="label_18">
          <property name="text">
           <string>Command line</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="commandLineEdit"/>
        </item>
        <item>
         <widget class="QLabel" name="label_19">
          <property name="text">
           <string>Console output</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="consoleEdit"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="pushButton_run">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Classify with current settings (check Console for output)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>Run</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_restore">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;reset all parameters to default&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>Restore defaults</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>628</width>
     <height>27</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionInput"/>
    <addaction name="actionTraining"/>
    <addaction name="actionMask"/>
    <addaction name="actionOutput"/>
    <addaction name="actionColor_table"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionInput">
   <property name="text">
    <string>Input</string>
   </property>
   <property name="shortcut">
    <string>Alt+I</string>
   </property>
  </action>
  <action name="actionTraining">
   <property name="text">
    <string>Training</string>
   </property>
   <property name="shortcut">
    <string>Alt+T</string>
   </property>
  </action>
  <action name="actionMask">
   <property name="text">
    <string>Mask</string>
   </property>
   <property name="shortcut">
    <string>Alt+M</string>
   </property>
  </action>
  <action name="actionOutput">
   <property name="text">
    <string>Output</string>
   </property>
   <property name="shortcut">
    <string>Alt+O</string>
   </property>
  </action>
  <action name="actionColor_table">
   <property name="text">
    <string>Color table</string>
   </property>
   <property name="shortcut">
    <string>Alt+C</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>training</tabstop>
  <tabstop>toolButton_training</tabstop>
  <tabstop>tln</tabstop>
  <tabstop>label</tabstop>
  <tabstop>nclass</tabstop>
  <tabstop>input</tabstop>
  <tabstop>toolButton_input</tabstop>
  <tabstop>mask</tabstop>
  <tabstop>toolButton_mask</tabstop>
  <tabstop>msknodata</tabstop>
  <tabstop>output</tabstop>
  <tabstop>toolButton_output</tabstop>
  <tabstop>nodata</tabstop>
  <tabstop>ct</tabstop>
  <tabstop>toolButton_ct</tabstop>
  <tabstop>svmtype</tabstop>
  <tabstop>coef0</tabstop>
  <tabstop>nu</tabstop>
  <tabstop>kerneltype</tabstop>
  <tabstop>kd</tabstop>
  <tabstop>gamma</tabstop>
  <tabstop>ccost</tabstop>
  <tabstop>cv</tabstop>
  <tabstop>commandLineEdit</tabstop>
  <tabstop>consoleEdit</tabstop>
  <tabstop>tableView_labels</tabstop>
  <tabstop>commandLinkButtonPrepareTable</tabstop>
  <tabstop>pushButton_run</tabstop>
  <tabstop>pushButton_restore</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
